ansible/roles/airflow/tasks/main.yml

```diff
@@ -0,0 +1,155 @@
+    restart_retries: 2
+
+- name: shared airflow config templates
+  file: dest=/srv/etc/af-share-{{ airflow_tag }} state=directory owner=root group=root mode=0750
+- stat: path=/srv/etc/af-share-{{ airflow_tag }}/passwd
+  register: afpasswd
+- stat: path=/srv/etc/af-share-{{ airflow_tag }}/group
+  register: afgroup
+- stat: path=/srv/etc/af-share-{{ airflow_tag }}/airflow.cfg
+  register: afcfg
+
+- name: create temporary openobservatory/airflow container to extract file
+  docker_container:
+    image: openobservatory/airflow:{{ airflow_tag }}
+    name: af-tmp
+    purge_networks: true
+    state: present # NB: it's not `started`
+  when: not (afpasswd.stat.exists and afgroup.stat.exists and afcfg.stat.exists)
+- command: docker cp af-tmp:/etc/passwd /srv/etc/af-share-{{ airflow_tag }}/passwd
+  when: not afpasswd.stat.exists
+- command: docker cp af-tmp:/etc/group /srv/etc/af-share-{{ airflow_tag }}/group
+  when: not afgroup.stat.exists
+- command: docker cp af-tmp:/usr/local/airflow/airflow.cfg /srv/etc/af-share-{{ airflow_tag }}/airflow.cfg
+  when: not afcfg.stat.exists
+- name: drop temporary openobservatory/airflow container
+  docker_container:
+    name: af-tmp
+    state: absent
+
+- file: path=/srv/etc/af-share-{{ airflow_tag }}/airflow.cfg mode=0644 owner=root group=root
+
+- lineinfile:
+    dest: /srv/etc/af-share-{{ airflow_tag }}/group
+    line: "{{ item.login }}:x:{{ item.id }}:"
+  with_items:
+    - "{{ passwd.afsched }}"
+    - "{{ passwd.afwebsrv }}"
+    - "{{ passwd.afworker }}"
+
+- lineinfile:
+    dest: /srv/etc/af-share-{{ airflow_tag }}/passwd
+    line: "{{ item.login }}:x:{{ item.id }}:{{ item.id }}:{{ item.comment }}:/usr/local/airflow:/bin/bash"
+  with_items:
+    - "{{ passwd.afsched }}"
+    - "{{ passwd.afwebsrv }}"
+    - "{{ passwd.afworker }}"
+
+- name: mkdir etc/af-worker
+  file: dest=/srv/etc/af-worker state=directory owner=root group=root mode=0700
+- template: dest=/srv/etc/af-worker/s3read.env src=af-worker-s3read.env owner=root group=root mode=0400
+
+- name: mkdir af-worker-bin
+  file: dest=/srv/af-worker-bin state=directory owner=root group=root mode=0755
+- name: copy docker-trampoline
+  tags: airflow-dags
+  copy: src=docker-trampoline dest=/srv/af-worker-bin/ owner=root group=root mode=0755
+  notify: restart af-worker # because the _file_ is bind-monted, not a directory
+
+- name: mkdir airflow-dags
+  file: dest=/srv/airflow-dags state=directory owner=root group=root mode=0755
+- name: copy airflow-dags # depends on `rsync`
+  tags: airflow-dags
+  synchronize: src=airflow-dags/ dest=/srv/airflow-dags/ archive=no recursive=yes checksum=yes delete=yes
+  notify: restart airflow
+
+- include: af-container.yml dockname=af-sched  login=afsched  command=scheduler
+- include: af-container.yml dockname=af-websrv login=afwebsrv command=webserver ipv4_address=172.26.43.254
+- include: af-container.yml dockname=af-worker login=afworker command=worker image_ext=-worker
```

Comment:
Well according to Ansible's BCP and role organization and sanity of the people that would like to (re)build a similar setup it will be "nicer" IMHO to have the include in the main as separate tasks.